src/azul/terraform.py

```diff
@@ -290,6 +290,27 @@ def provider_fragment(region: str) -> JSON:
         return {'provider': f'aws.{region}'}
 
 
+def provision_s3_bucket_public_access_block(tf_config: dict) -> JSON:
+    """
+    Return a public access block for each of buckets defined in the Terraform
+    configuration JSON.
+    """
+    tf_config['resource']['aws_s3_bucket_public_access_block'] = {
```

Comment:
You need to decide if you want to modify in place, in which case you don't need to return anything, or if you want to make a copy, potentially a shallow one, and modify the copy. The documentation should be crystal-clear about that.